src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -85,20 +74,26 @@ class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>
 		m_dim_output = dims;
 	}
 
+	void set_kernel(const std::shared_ptr<Kernel>& kernel)
+	{
+		require(
```

Comment:
btw you can just make the argument ShiftInvariantKernel here and remove the runtime check. easy :)